File: be/Models/Image.cs
namespace be.Models
{
    public class Image
    {
        public int ImageID { get; set; } // Primary key, INTEGER, auto-increment
        public string ImageUrl { get; set; } // TEXT, not null
        public string ReferenceType { get; set; } // STRING(20), not null
        public string ReferenceID { get; set; } // STRING(10), not null

        // Quan hệ
        public Product? Product { get; set; }
    }
}

File: be/src/models/Image.js
'use strict';
const { Model } = require('sequelize');

module.exports = (sequelize, DataTypes) => {
  class Image extends Model {
    static associate(models) {
      Image.belongsTo(models.Product, { foreignKey: 'ReferenceID', constraints: false, scope: { ReferenceType: 'Product' } });
      Image.belongsTo(models.Appointment, { foreignKey: 'ReferenceID', constraints: false, scope: { ReferenceType: 'Appointment' } });
    }
  }

  Image.init(
    {
      ImageID: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true,
        allowNull: false,
      },
      Image: {
        type: DataTypes.TEXT,
        allowNull: false,
      },
      ReferenceType: {
        type: DataTypes.STRING(20),
        allowNull: false,
      },
      ReferenceID: {
        type: DataTypes.STRING(10),
        allowNull: false,
      },
    },
    {
      sequelize,
      modelName: 'Image',
      tableName: 'Image',
      timestamps: false,
      indexes: [
        { fields: ['ReferenceType', 'ReferenceID'], name: 'index_reference' },
      ],
    }
  );

  return Image;
};